### PR TITLE
Remove outdated tau code

### PR DESCRIPTION
**Pull request by steggema**
#### PR description:

This removes a lot of outdated and unused tau code, in particular the full package RecoTauTag/TauTagTools and the CaloTau reconstruction. The goal is do make tau development easier and to reduce the maintenance burden. No changes are expected in any of the workflows.

#### PR validation:

Matrix tests were run locally as well as the tau reconstruction sequence itself to validate that there are no unwanted effects. The changes were reviewed by the RECO contacts @mbluj @swozniewski, also to confirm that there are no customers outside of CMSSW to the best of our knowledge.

@rmanzoni @roger-wolf

### CODE REVIEW

**Comment by perrotta** [comment]
please test

**Comment by steggema** [comment]
For what it's worth, I'm not sure why this didn't error didn't appear in a local build following the regular instructions here http://cms-sw.github.io/PRWorkflow.html (neither for me nor for our RECO contacts), so I hope there aren't more of those.

**Comment by perrotta** [comment]
please test

**Comment by steggema** [comment]
The new header consistency error 
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cf1b76/299/headers_chks.log
seems to come from something not related to this PR?

**Comment by perrotta** [comment]
please test

File: RecoTauTag/RecoTau/interface/TauTagTools.h [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#ifndef RecoTauTag_RecoTau_TauTagTools_h
+#define RecoTauTag_RecoTau_TauTagTools_h
+
+#include "DataFormats/Candidate/interface/Candidate.h"
+#include "DataFormats/Candidate/interface/CandidateFwd.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
+#include "DataFormats/VertexReco/interface/VertexFwd.h"
+
+namespace TauTagTools{
```
Comment (slava77) [inline RecoTauTag/RecoTau/interface/TauTagTools.h]: namespaces should be all in lower case 
http://cms-sw.github.io/cms_coding_rules.html#2--naming-rules-1

**Comment by kpedro88** [comment]
+upgrade
removed unneeded dependency in TrackTrigger package

File: RecoTauTag/RecoTau/interface/TauTagTools.h (hunk 1 above)
Comment (steggema) [reply RecoTauTag/RecoTau/interface/TauTagTools.h]: It was mostly a simple copy from the removed TauTagTools folder (removing unneeded code in this file), but I fixed it nonetheless.

**Comment by perrotta** [comment]
please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/26816	ccfac0e
- code changes are in line with the PR description and the follow up review. 
    - This is a technical PR (no physics results in production setup is affected)
- jenkins tests pass and comparisons with the baseline show no differences

**Comment by kpedro88** [comment]
+upgrade

**Comment by santocch** [comment]
+1

**Comment by fabiocos** [comment]
@rekovic the change for l1 is trivial and just in 1 file
@Dr15Jones @smuzaffar for core this just remove the package from the Utilities list

**Comment by smuzaffar** [comment]
+core

**Comment by fabiocos** [comment]
+1

@rekovic the L1 part looks pretty minimal, please check and comment in case, I move forward with the integration

**Comment by fabiocos** [comment]
merge

**Pull request merged**